src/ctelnet.cpp

```diff
@@ -1267,6 +1281,14 @@ void cTelnet::processTelnetCommand(const std::string& command)
 #ifdef DEBUG_TELNET
         qDebug().nospace().noquote() << "Server sent telnet IAC DO " << decodeOption(option);
 #endif
+        if (option == OPT_CHARSET) {
+            // CHARSET support per https://tools.ietf.org/html/rfc2066
+            enableCHARSET = true;
+            sendTelnetOption(TN_DO, OPT_CHARSET);
```

Comment:
I saw the his/our/announced mappings and spent some time trying to make sense of them.  The naming and comments there need some work.  It feels like something was missing from one of the perspectives.

Reply:
I *think* our code is based on "The Q Method of Implementing TELNET Option Negotiation" **[RFC 1143](https://tools.ietf.org/html/rfc1143)** but looks as though it might have been simplified because we do not tend to request things ourself (I could be wrong however).

According to section 7 of that RFC there is an alternative methodology based on a pair (one for each direction) of 6 state variables - but my enquires about it remain unanswered https://stackoverflow.com/q/45065674/4805858 .

Reply:
For what it is worth you might want to take a look at the corresponding code in KMuddy which we ~~shamelessly lifted~~acquired via GPL 2+ terms right at the start of the project: https://github.com/KDE/kmuddy/blob/master/kmuddy/ctelnet.cpp .